scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/Resources.java

```diff
@@ -63,16 +65,15 @@ private Resources() {
                 .build();
     }
 
-    public static List<Integer> selectTwoPortsFromRange(List<Protos.Resource> offeredResources) {
-        List<Integer> ports = new ArrayList<>();
-        offeredResources.stream().filter(resource -> resource.getType().equals(org.apache.mesos.Protos.Value.Type.RANGES))
-                .forEach(resource -> resource.getRanges().getRangeList().stream().filter(range -> ports.size() < 2).forEach(range -> {
-                    ports.add((int) range.getBegin());
-                    if (ports.size() < 2 && range.getBegin() != range.getEnd()) {
-                        ports.add((int) range.getBegin() + 1);
-                    }
-                }));
-        return ports;
+    public static List<Integer> selectTwoUnprivilegedPortsFromRange(List<Protos.Resource> offeredResources) {
+        return offeredResources.stream()
+                .filter(resource -> resource.getType().equals(Protos.Value.Type.RANGES))
+                .flatMap(resource -> resource.getRanges().getRangeList().stream())
+                .flatMapToInt(range -> IntStream.rangeClosed((int) range.getBegin(), (int) range.getEnd()))
+                .filter(port -> port > 1024)
```

Comment:
Are all ports below 1025 privileged? Logstash also relies on presence of port 5000: https://github.com/mesos/logstash/blob/master/logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/TaskInfoBuilder.java#L42

If 1024 is an agreed number, it deserves to become `static public final`